core/src/oracle_state.rs

```diff
@@ -68,6 +69,10 @@ pub trait StageDataSource {
 
 pub trait PoolBoxSource {
     fn get_pool_box(&self) -> Result<PoolBoxWrapper>;
+    /// Get Pool Box without any parameter/token checks
+    fn get_pool_box_raw(&self) -> Result<ErgoBox> {
```

Comment:
Could you please explain why it's needed? I checked its usage, and I don't get it.